src/reducer.ts

```diff
@@ -32,9 +32,9 @@ function isEmpty(obj: object) {
 
 export function formControlReducer<TValue extends SupportedNgrxFormControlValueTypes>(
   state: FormControlState<TValue>,
-  action: Action,
```

Comment:
The `action` is intentionally typed as `Action` since this is the public API and we need to fulfill the contract of `ActionReducer<FormControlState<TValue>>` which takes just an `Action`, not an `Actions<TValue>`. The internal functions are for stricter typing during testing.